subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio.c

```diff
@@ -2565,10 +2598,57 @@ void radio_ar_configure(uint32_t nirk, void *irk, uint8_t flags)
 
 	NRF_AAR->ENABLE = (AAR_ENABLE_ENABLE_Enabled << AAR_ENABLE_ENABLE_Pos) &
 			  AAR_ENABLE_ENABLE_Msk;
+
+#if defined(CONFIG_SOC_COMPATIBLE_NRF54LX)
+	aar_job.nirk = nirk;
+
+	/* Input, Resolvable Address Hash offset in the legacy or extended advertising PDU.
+	 * Radio packet pointer offset by 3 compared to legacy	NRF_AAR that took Radio packet
+	 * pointer value.
+	 */
+	aar_job.in[0].ptr = (uint8_t *)addrptr + 3U;
+	aar_job.in[0].length = 3U;
+	aar_job.in[0].attribute = AAR_JOB_PTR_ATTRIBUTE_HASH;
+
+	/* Input, Resolvable Address Random offset in the legacy or extended advertising PDU.
+	 * Radio packet pointer offset by 3 compared to legacy	NRF_AAR that took Radio packet
+	 * pointer, plus offset of the 24-bit random in the legacy or extended advertising PDU after
+	 * the 24-bit Hash in the Resolvable Address.
+	 */
+	aar_job.in[1].ptr = (uint8_t *)addrptr + 6U;
+	aar_job.in[1].length = 3U;
+	aar_job.in[1].attribute = AAR_JOB_PTR_ATTRIBUTE_PRAND;
+
+	/* Input, list of IRKs used for resolution */
+	for (uint32_t i = 0; i < nirk; i++) {
+		aar_job.in[2U + i].ptr = (void *)(((uint8_t *)irk) + (16U * i));
+		aar_job.in[2U + i].length = 16U;
+		aar_job.in[2U + i].attribute = AAR_JOB_PTR_ATTRIBUTE_IRK;
+	}
+
+	/* A terminating empty job entry */
+	aar_job.in[2U + nirk].ptr = 0U;
+	aar_job.in[2U + nirk].length = 0U;
+	aar_job.in[2U + nirk].attribute = 0U;
```

Comment:
The function does not protect against any issues where `nirk > CONFIG_BT_CTLR_RL_SIZE`

Reply:
Wouldn't it still make sense to add an assert or something similar, in case the upper layers provide an invalid `nirk` value?